tests/minimal_run.py

```diff
@@ -0,0 +1,33 @@
+import subprocess
+import psutil
+import time
+import json
+import urllib.request
+
+
+def main():
+    process = subprocess.Popen(
+        ["python", "tests/simple_server.py"],
+    )
+    print("Waiting for server to start...")
+    time.sleep(5)
+    try:
+        url = "http://127.0.0.1:8000/predict"
+        data = json.dumps({"input": 4.0}).encode("utf-8")
+        headers = {"Content-Type": "application/json"}
+        request = urllib.request.Request(url, data=data, headers=headers, method="POST")
+        response = urllib.request.urlopen(request)
+        status_code = response.getcode()
+        assert status_code == 200
+    except Exception:
+        raise
+
+    finally:
```

Comment:
You can probably use the decorator for this, the one we introduced in e2e tests.

Reply:
it would need to import from [tests](https://github.com/Lightning-AI/LitServe/blob/4a3721c89f44bc7ac2f0f98f147fa0c699e7878f/tests/e2e/test_e2e.py#L25) which includes additional package like PyTorch. I can keep that a separate utility module but at this time for only this test it seemed overkill. Maybe we can do that if when we need to to use that decorator in multiple modules?

Reply:
ok sounds good